go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+		{
+			desc:             "Given address is present on the link",
+			inputLink:        mockLink,
+			inputAddrToMatch: ovntest.MustParseIPNet("192.168.1.15/24"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{
+					"AddrList",
+					[]string{"*mocks.Link", "int"},
+					[]interface{}{
+						[]netlink.Addr{
+							{
+								IPNet: ovntest.MustParseIPNet("192.168.1.15/24"),
+							},
+						},
+						nil,
+					},
+				},
+			},
+		},
+		{
+			desc:             "Given address is NOT present on the link",
+			inputLink:        mockLink,
+			inputAddrToMatch: ovntest.MustParseIPNet("192.168.1.15/24"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrList", []string{"*mocks.Link", "int"}, []interface{}{nil, nil}},
+			},
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			for _, item := range tc.onRetArgsLinkIfaceOpers {
+				call := mockLink.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+			flag, err := LinkAddrExist(tc.inputLink, tc.inputAddrToMatch)
+			t.Log(flag, err)
+			if tc.errExp {
+				assert.Error(t, err)
+			} else {
+				assert.Nil(t, err)
+			}
+			mockNetLinkLibOperations.AssertExpectations(t)
+			mockLink.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLinkAddrAdd(t *testing.T) {
+	mockNetLinkLibOperations := new(mocks.NetLinkLibOperations)
+	mockLink := new(netlink_mocks.Link)
+	// below is defined in net_linux.go
+	netLinkOperations = mockNetLinkLibOperations
+
+	tests := []struct {
+		desc                     string
+		inputLink                netlink.Link
+		inputNewAddr             *net.IPNet
+		errExp                   bool
+		onRetArgsNetLinkLibOpers []onCallReturnArgs
+		onRetArgsLinkIfaceOpers  []onCallReturnArgs
+	}{
+		// TODO: Code may need to be updated to check that link is not-nil to avoid panic
+		/*{
+			desc: "link is nil",
+			inputLink: nil,
+			inputNewAddr: nil,
+			errExp: true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{nil, fmt.Errorf("mock error")}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{},[]interface{}{&netlink.LinkAttrs{Name: "testIfaceName",}}},
+			},
+		},*/
+		{
+			desc:         "setting <nil> address on link errors out",
+			inputLink:    mockLink,
+			inputNewAddr: nil,
```

Comment:
are you missing `errExp: true,`?

Reply:
I believe so, let me confirm and fix accordingly